build/Microsoft.DotNet.Cli.Prepare.targets

```diff
@@ -1,47 +1,98 @@
+  <UsingTask TaskName="PrepareTargets" AssemblyFile="$(CLIBuildDll)" />
+  <UsingTask TaskName="GetCurrentRuntimeInformation" AssemblyFile="$(CLIBuildDll)" />
   <UsingTask TaskName="GenerateBuildVersionInfo" AssemblyFile="$(CLIBuildDll)" />
+  <UsingTask TaskName="CheckPrereqs" AssemblyFile="$(CLIBuildDll)" />
+  <UsingTask TaskName="SetEnvVar" AssemblyFile="$(CLIBuildDll)" />
+
+  <Target DependsOnTargets="BuildDotnetCliBuildFramework" Name="Prepare">
+    <PrepareTargets />
+  </Target>
 
   <Target Name="Init" >
     <!-- Current Runtime Information -->
-    <Rid>
-      <Output TaskParameter="OutputRid" PropertyName="Rid" />
-    </Rid>
-    <Architecture>
-      <Output TaskParameter="OutputArchitecture" PropertyName="Architecture" />
-    </Architecture>
+    <GetCurrentRuntimeInformation>
+      <Output TaskParameter="Rid" PropertyName="Rid" />
+      <Output TaskParameter="Architecture" PropertyName="Architecture" />
+      <Output TaskParameter="OSName" PropertyName="OSName" />
+    </GetCurrentRuntimeInformation>
 
     <!-- Common Properties -->
     <PropertyGroup>
       <BaseOutputDirectory>$(RepoRoot)/artifacts/$(Rid)</BaseOutputDirectory>
       <OutputDirectory>$(BaseOutputDirectory)/stage2</OutputDirectory>
       <Stage2CompilationDirectory>$(BaseOutputDirectory)/stage2compilation</Stage2CompilationDirectory>
       <IntermediateDirectory>$(BaseOutputDirectory)/intermediate</IntermediateDirectory>
+      <PackagesDirectory>$(BaseOutputDirectory)/packages</PackagesDirectory>
     </PropertyGroup>
 
-    <!-- Generates BuildVersion Item
-        BuildVersion Metadata:
-          - Major
-          - Minor
-          - Patch
-          - ReleaseSuffix
-          - CommitHash
-          - CommitCount
-          - VersionSuffix
-          - SimpleVersion
-          - NugetVersion
-          - MsiVersion            -->
     <GenerateBuildVersionInfo RepoRoot="$(RepoRoot)">
-        <Output TaskParameter="OutputBuildVersionInfo"
-                ItemName="BuildVersion" />
+      <Output TaskParameter="VersionMajor" PropertyName="VersionMajor" />
+      <Output TaskParameter="VersionMinor" PropertyName="VersionMinor" />
+      <Output TaskParameter="VersionPatch" PropertyName="VersionPatch" />
+      <Output TaskParameter="CommitHash" PropertyName="CommitHash" />
+      <Output TaskParameter="CommitCount" PropertyName="CommitCount" />
+      <Output TaskParameter="ReleaseSuffix" PropertyName="ReleaseSuffix" />
+      <Output TaskParameter="VersionSuffix" PropertyName="VersionSuffix" />
+      <Output TaskParameter="SimpleVersion" PropertyName="SimpleVersion" />
+      <Output TaskParameter="NugetVersion" PropertyName="NugetVersion" />
+      <Output TaskParameter="MsiVersion" PropertyName="MsiVersion" />
+      <Output TaskParameter="VersionBadgeMoniker" PropertyName="VersionBadgeMoniker" />
+      <Output TaskParameter="Channel" PropertyName="Channel" />
```

Comment:
I would eventually move all these properties directly into a props file instead of needing to read them from the test file and pull them in this way. I understand it probably makes sense to keep them this way while the other build system is still in play and also reads the text file.